src/pal/src/include/pal/threadsusp.hpp

```diff
@@ -73,6 +73,8 @@ Module Name:
 #include <sys/sem.h>
 #include <sys/types.h>
 
+#elif __ANDROID__
+#include <semaphore.h>
```

Comment:
@am11 Would love to do this, but HAVE_SEMAPHORE_H is not defined by default.
I added a check to `src/pal/src/configure.cmake` like this:

```
check_include_files(semaphore.h HAVE_SEMAPHORE_H)
```

but somehow this is never evaluated.

I feel I'm missing something obvious, so if you can help :)